src/ui/actions/timeline.ts

```diff
@@ -500,3 +516,40 @@ export function clearHoveredItem(): UIThunkAction {
 export function setPlaybackPrecachedTime(time: number): SetPlaybackPrecachedTimeAction {
   return { type: "set_playback_precached_time", time };
 }
+
+export function setTrimRegion(trimRegion: {
+  startTime: number;
+  endTime: number;
+}): SetTrimRegionAction {
+  return { type: "set_trim_region", trimRegion };
+}
+
+export function updateTrimRegion(
+  type: "startTime" | "endTime" | "span",
+  midpointDifference: number
+): UIThunkAction {
+  return ({ dispatch, getState }) => {
+    const state = getState();
+    const hoverTime = getHoverTime(state);
+    const trimRegion = getTrimRegion(state);
+    const zoomRegion = getZoomRegion(state);
+
+    if (!hoverTime || !trimRegion) return;
```

Comment:
I would maybe word this as `if !(hoverTime && trimRegion) return;`
Just to reduce the number of negatives, but either way it's a little hard to parse. Maybe give this a meaningful name? What do both of these values being true indicate? What does it mean if one or both of them is missing?

Reply:
Also, it looks like we also use this check elsewhere on line 126 of trimmer and maybe also on line 10 of tooltip. Maybe we should centralize that logic somewhere?